frontend/src/app/components/channel/channel.singleUser.controller.ts

```diff
@@ -41,20 +41,26 @@ export class SingleUserChannelController extends BaseChannelController {
     }
 
     var oldProgramUser = this.programUser;
-    this.programUser = user;
 
     if (user === null) {
       this.slotElement.program.src = null;
+      this.programUser = null;
     } else {
-      attachMediaStream(this.slotElement.program, this.programUser.stream);
-
-      this.forwardProgramToSDI();
+      this.programUser = user;
+
+      this.forwardProgramToSDI().then(() => {
+        attachMediaStream(this.slotElement.program, this.programUser.stream);
+
+        if (oldProgramUser) {
+          this.videoRoom.unsubscribeFromStream(oldProgramUser.login);
+          oldProgramUser.stream = null;
+          this.$log.debug('Unsubscribed from', oldProgramUser.login);
+        }
+      }, () => {
+        // Rolling back current program user in case of error
+        this.programUser = oldProgramUser;
```

Comment:
it does call it inside forwardProgramToSDI before propagating to here
add one more report here?